sdk/core/Azure.Core/src/Pipeline/RetryPolicy.cs

```diff
@@ -127,7 +122,13 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
 
                 if (shouldRetry)
                 {
-                    TimeSpan delay = async ? await CalculateNextDelayAsync(message).ConfigureAwait(false) : CalculateNextDelay(message);
+                    if (message.RetryNumber == 0)
+                    {
+                        message.SetProperty(typeof(RetryPolicyPropertiesKey), new Dictionary<string, object?>());
```

Comment:
Is this intended to be a generic property bag to be used by implementors that is opaque to us?